private/GetScriptFunctions.ps1

```diff
@@ -0,0 +1,56 @@
+﻿function GetScriptFunctions {
+    <#
+        .SYNOPSIS
+            Outputs all functions within a Script or ScriptBlock.
+
+        .DESCRIPTION
+            Uses AST to retrieve all functions from a valid script file, 
+            then outputs them all. Useful for dot-sourcing functions from 
+            a file that also contains running code.
+
+        .EXAMPLE
+            GetScriptFunctions -Export -ScriptPath .\testscript.ps1 | iex
+
+        .EXAMPLE
+            (GetScriptFunctions -ScriptPath .\testscript.ps1).Count
+            7
+
+        .EXAMPLE
+            Get-ChildItem -Recurse -Filter *.ps1 | GetScriptFunctions | Set-Content 'functions.psm1'
+    #>
+    [OutputType([System.Management.Automation.Language.ScriptBlockAst[]])]
+    [OutputType([System.Management.Automation.Language.FunctionDefinitionAst[]])]
+    param(
+        # FilePath for script to analyse
+        [Parameter(Mandatory, ValueFromPipelineByPropertyName, ValueFromPipeline, Position = 0)]
+        [ValidateScript( {Test-Path $_})]
+        [Alias('FullName')]
+        [IO.FileInfo[]]$ScriptPath,
+
+        # To export the function(s) extent alone, or with full AST data
+        [Switch]$Export
+    )
+    process {
+        foreach ($Item in $ScriptPath) {
```

Comment:
You **need** to `Convert-Path` here (on the right-hand side of `in`) because:

1. ScriptPath could be relative
1. ScriptPath could be a PowerShell provider path like `FileSystem::\Questionmark\`
1. Each ScriptPath could have wildcards and refer to many files

Reply:
Done